File: Signiture-Backend/kotlin/delivery/api/src/main/java/pcy/study/api/domain/token/business/TokenBusiness.java
package pcy.study.api.domain.token.business;

import lombok.RequiredArgsConstructor;
import pcy.study.common.annotation.Business;
import pcy.study.common.api.code.ErrorCode;
import pcy.study.common.exception.ApiException;
import pcy.study.api.domain.token.controller.model.TokenResponse;
import pcy.study.api.domain.token.converter.TokenConverter;
import pcy.study.api.domain.token.service.TokenService;

import java.util.Objects;
import java.util.UUID;

@Business
@RequiredArgsConstructor
public class TokenBusiness {

    private final TokenService tokenService;
    private final TokenConverter tokenConverter;

    public TokenResponse issueToken(Long userId) {
        Objects.requireNonNull(userId, () -> {
            throw new ApiException(ErrorCode.NULL_POINT, "User ID is Null");
        });

        var accessToken = tokenService.issueAccessToken(userId);
        var randomToken = UUID.randomUUID().toString();
        var refreshToken = tokenService.issueRefreshToken(randomToken);
        return tokenConverter.toResponse(accessToken, refreshToken);
    }

    public Long validationAccessToken(String accessToken) {
        return tokenService.validationAccessToken(accessToken);
    }
}

File: Signiture-Backend/kotlin/delivery/account/src/main/kotlin/pcy/study/account/domain/token/business/TokenBusiness.kt
package pcy.study.account.domain.token.business

import pcy.study.account.domain.token.controller.model.TokenResponse
import pcy.study.account.domain.token.controller.model.TokenValidationResponse
import pcy.study.account.domain.token.converter.TokenConverter
import pcy.study.account.domain.token.service.TokenService
import pcy.study.common.annotation.Business
import java.util.*

@Business
class TokenBusiness(
    private val tokenService: TokenService,
    private val tokenConverter: TokenConverter
) {

    fun issueToken(userId: Long): TokenResponse {
        val accessToken = tokenService.issueAccessToken(userId)
        val randomToken = UUID.randomUUID().toString()
        val refreshToken = tokenService.issueRefreshToken(randomToken)
        return tokenConverter.toTokenResponse(accessToken, refreshToken)
    }

    fun validationAccessToken(accessToken: String): TokenValidationResponse {
        val userId = tokenService.validationAccessToken(accessToken)
        return tokenConverter.toTokenValidationResponse(userId)
    }
}